core/matchdata/deconstruct_keys_spec.rb

```diff
@@ -0,0 +1,52 @@
+require_relative '../../spec_helper'
+
+describe "MatchData#deconstruct_keys" do
+  ruby_version_is "3.2" do
+    it "returns whole hash for nil as an argument" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.deconstruct_keys(nil).should == { f: "foo", b: "bar" }
+    end
+
+    it "returns only specified keys" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.deconstruct_keys([:f]).should == { f: "foo" }
+    end
+
+    it "requires one argument" do
+      -> {
+        /l/.match("l").deconstruct_keys
+      }.should raise_error(ArgumentError)
+    end
+
+    it "it raises error when argument is neither nil nor array" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      -> { m.deconstruct_keys(1) }.should raise_error(TypeError, "wrong argument type Integer (expected Array)")
+      -> { m.deconstruct_keys("asd") }.should raise_error(TypeError, "wrong argument type String (expected Array)")
+      -> { m.deconstruct_keys(:x) }.should raise_error(TypeError, "wrong argument type Symbol (expected Array)")
+      -> { m.deconstruct_keys({}) }.should raise_error(TypeError, "wrong argument type Hash (expected Array)")
+    end
+
+    it "returns {} when passed []" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      m.deconstruct_keys([]).should == {}
+    end
+
+    it "does not accept non-Symbol keys" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+      -> {
+        m.deconstruct_keys(['year', :foo])
+      }.should raise_error(TypeError)
+    end
+
+    it "ignores not existing Symbol keys" do
+      m = /(?<f>foo)(?<b>bar)/.match("foobar")
+
+     m.deconstruct_keys([:f, :a]).should == { f: "foo" }
+    end
```

Comment:
**minor:** it doesn't just ignore not existing keys. It stops iterating and doesn't process any key after not-existing one.